### ISSUE TITLE
[Cross-Project Epic] CI Build / Weekly Release process should use consistent quay tags for sha-version/next and 7.yy.z/latest

### ISSUE THREAD

**Issue by nickboldt** [eclipse-che/che#19291]
### Is your task related to a problem? Please describe.

Today we use a harvest medley of terms to describe daily and continuous integration builds across the various Che projects. Some examples include:

* [Che Theia](https://quay.io/repository/eclipse/che-theia?tag=latest&tab=tags) uses **next** to mean continuous integration builds; also pushes a sha-versioned tag

* [Che Dashboard](https://quay.io/repository/eclipse/che-dashboard?tag=latest&tab=tags) used to use **next-react** to mean "a huge refactoring project that isn't stable yet and will eventually replace the current dashboard, but needs its own fork/branch/tag stream", and **next** to mean continuous integration builds; does NOT push sha-versioned tags

* [Che server](https://quay.io/repository/eclipse/che-server?tag=latest&tab=tags) uses **latest** for releases, and **nightly** for a daily build (last one 14 hrs ago, but last commits 8 hrs ago); does NOT push sha-versioned tags
  * [Rename/replace nightly build of che-server with on-push workflow #19894](https://github.com/eclipse/che/issues/19894)

* [Che machine exec](https://quay.io/repository/eclipse/che-machine-exec?tag=latest&tab=tags) uses **nightly** to mean continuous integration builds; also pushes a sha-versioned tag

* [Che devfile registry](https://quay.io/repository/eclipse/che-devfile-registry?tag=latest&tab=tags) uses **nightly** to mean continuous integration builds; also pushes a sha-versioned tag

### Describe the solution you'd like

Change to a more consistent naming convention for continuous integration builds, large refactorings, and releases:

* ~~"ci" for "continuously building new images on every commit"~~
* "next" for "continuously building new images on every commit"
* ~~"next" for special case refactorings (eg., Dashboard Next or Plugin Registry 3.0)~~
* "nightly" is dead, won't be used anymore as we won't do daily/nightly builds (only CI builds)
* optional "latest" for pointer to 7.yy.z latest released container images (where / if useful)
* sha-versioned tags for all releases and CI builds

In addition to impacting GH action workflows to use different tags, this will impact:

* plugin registry references to :next or :nightly: https://github.com/eclipse/che-plugin-registry/blob/master/che-editors.yaml
* PR checks
* E2E tests

### Describe alternatives you've considered

Enjoy the multicultural experience that is guessing what each project means by "nightly", "next", or "latest".

**Comment by tsmaeder** [comment]
> "next" for special case refactorings (eg., Dashboard Next or Plugin Registry 3.0)

I don't think it makes sense to have a special tag for those. For starters, there might be multiple of those refactorings going on in parallel (next1, next2)? And second, I think a format like "dashboard_react_ci", etc. would be more descriptive and allow use to have a whole parallel ci/latest/sha, etc. tag stream going.

**Comment by nickboldt** [comment]
Right but my point is that TODAY, "next" has special meaning and so should not be considered the same as "nightly", when only ONE project uses it for ci builds (che-theia), while everyone else uses nightly. 

And so as to offend everyone equally, we can move to "ci" for ALL nightly/next/ci builds, rather than a mix of terms.

**Comment by benoitf** [comment]
I think nobody understand `:ci` (as all builds are built by CI anyway (nightlies, per commits, releases, etc. )

Also `next`is not only used by theia or che-theia, it's used by all npmjs projects anyway: 
https://docs.npmjs.com/about-npm-versions#the-next-release-of-npm

**Comment by nickboldt** [comment]
Forgot about this thread, so getting back to it now. 

* releases are not ci. They are not continuous integration. They are on-request, triggered on a specific date, by a human.

* ci builds (call 'em nightly. integration, per commit, whatever) are continuously built on commits, and are thus CI.

* if `next` is a convention used by npm projects, and Che is an amalgam of golang, bash, npm, typescript.... could we pick a label that's consistent across the whole project? (eg., machine exec and theia and dashboard all use different tags)? Why make things needlessly complicated? Do we WANT the community to be able to easily contribute, or are we intentionally obfuscating and treating each subproject like its own fiefdom?

**Comment by benoitf** [comment]
sorry but I find `ci` counter intuitive about what it contains and when it's build

**Comment by benoitf** [comment]
having chectl with two channels, one being `next` and another being `stable` I would opt-in for `next` tags on images

**Comment by nickboldt** [comment]
If everyone is good with :next then I'm happy to start moving the :nightly tags to :next so we're consistent across all of Che.

(Meanwhile in operatorhub, the convention is :preview but hey. :) )

**Comment by nickboldt** [comment]
I've updated my original proposal. How do we like this?

![image](https://user-images.githubusercontent.com/227597/116702843-9337de00-a997-11eb-8d5d-e2321d546f79.png)

**Comment by benoitf** [comment]
👍 

regarding another topic about multi-arch (and travis usage or not) maybe we'll end up with images having arch suffix as well

I like also about having also sha-versioned tags (it was already there in your proposal)
sometimes current images are broken and it's impossible to get back in time.
Maybe it should be easy to go back in days as well :D like `chectl update yesterday / chectl server:deploy`

**Comment by nickboldt** [comment]
Re: the chectl rollback to specific-version (or relative-date) ... probably useful. I've linked to the related rollback issues above.

**Comment by dmytro-ndp** [comment]
`:latest` is quite useful for testing pipelines (e.g. che-theia PR check), because we shouldn't change test script to reference latest Che version tag after each release.

**Comment by sleshchenko** [comment]
Note about:

> "next" for "continuously building new images on every commit"

some projects pulls resources from other project, like che-operator(devworkspace, devworkspace-che-operators), chectl (helm charts, che operator deployment).
So, for them `on every commit` may be not enough. We should keep timeline build (like each hour, nightly) and maybe even an ability to trigger commit manually, when person know that without it Che is going to fail.

**Comment by benoitf** [comment]
for pulling resources, we can send triggers to other projects as well
 like: new che-server commit : trigger a new chectl release to include these new deployments

**Comment by nickboldt** [comment]
+1, probably better to have a rebuild of che-operator trigger a rebuild of chectl (as I do for CRW) than to have hourly respins of the downstream project. Where we need to link projects, it's very easy to have one workflow trigger a downstream one. Can steal code from https://github.com/eclipse/che-release/blob/master/make-release.sh#L189-L191 to make that happen

**Comment by benoitf** [comment]
maybe we can use https://github.com/marketplace?type=actions&query=dispatch action as we might already be inside github actions anyway

https://github.com/marketplace/actions/repository-dispatch seems to have the most number of stars

**Comment by mkuznyetsov** [comment]
eclipse/che-devfile-registry
eclipse/che-plugin-registry
eclipse/che-server
eclipse/che
eclipse/che-machine-exec

the following projects now have `next` tagged builds (on latest commits), which also make SHA based tags

**Comment by nickboldt** [comment]
Thanks for the update! 

So that means we still have to convert these remaining projects:
* [ ] operator
* [ ] dashboard
* [ ] DWO
* [ ] DWCO
* [ ] image puller
* [ ] jwtproxy
* [ ] plugin broker images (2)
* [ ] theia dev, theia, and endpoint images (3)

**Comment by nickboldt** [comment]
Slip to 7.33; remaining TODOs:

- [ ] operator
- [ ] DWCO
- [ ] image puller

**Comment by nickboldt** [comment]
Remaining TODOs:

- [ ]  operator - eclipse-che/che-operator#863
- [ ] KIP - https://github.com/eclipse/che/issues/20024
- [ ] remove nightlies from all other projects (only :next should remain)

**Comment by nickboldt** [comment]
Starting this week, we'll start removing the :nightly tags from the release process and from quay.

**Comment by nickboldt** [comment]
Status:
- [x] configbump - n/a
- [ ] operator
- [ ] dashboard
- [x] devfile reg
- [ ] devfile reg sidecars
- [ ] intellij sidecar
- [ ] DWO
- [ ] DWCO
- [ ] KIP, KIPO
- [ ] jwtproxy 
- [x] machine-exec
- [x] plugin brokers - n/a
- [x] plugin reg
- [x] che server
- [ ] theia dev, theia, endpoint, vsix installer

**Comment by nickboldt** [comment]
Done: 
* KIP
* Dashboard
* DWCO

Still to do for consistency:

- [ ] https://github.com/eclipse/che-jwtproxy/blob/main/.github/workflows/build.yml#L81 uses :latest but no :next ; rename to `next-build.yml`
- [ ] intellij sidecar - rename https://github.com/che-incubator/jetbrains-editor-images/blob/main/.github/workflows/master.yml
- [ ] DWO - rename https://github.com/devfile/devworkspace-operator/blob/main/.github/workflows/dockerimage-next.yml to be consistent w/ other repos
- [ ] KIPO - https://github.com/eclipse/che/issues/20149

* [devfile reg sidecars](https://github.com/eclipse-che/che-devfile-registry/tree/main/arbitrary-users-patch) - delete old nightly image tags from quay

**Comment by ibuziuk** [comment]
regarding https://github.com/eclipse/che/issues/20149
@mkuznyetsov @nickboldt do you plan to provide an MR with the new CI based on the GH actions to  https://github.com/che-incubator/kubernetes-image-puller-operator ?
